### PR TITLE
Enable VPC cluster to accept a dedicated host pool ID and query it

### PR DESCRIPTION
**Pull request by KerDav**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
❯ make testacc TEST=./ibm/service/kubernetes TESTARGS='-run=TestAccIBMContainerVpcClusterDedicatedHost'
=== RUN   TestAccIBMContainerVpcClusterDedicatedHost
--- PASS: TestAccIBMContainerVpcClusterDedicatedHost (2986.71s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kubernetes	2987.721s

❯ make testacc TEST=./ibm/service/kubernetes TESTARGS='-run=TestAccIBMContainerVPCClusterDataSource_DedicatedHost'
=== RUN   TestAccIBMContainerVPCClusterDataSource_DedicatedHost
--- PASS: TestAccIBMContainerVPCClusterDataSource_DedicatedHost (2145.10s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kubernetes	2146.143s
```

### CODE REVIEW

File: ibm/acctest/acctest.go [hunk 1]
```diff
@@ -877,6 +880,11 @@ func init() {
 	if ClusterName == "" {
 		fmt.Println("[INFO] Set the environment variable IBM_CONTAINER_CLUSTER_NAME for ibm_container_nlb_dns resource or datasource else tests will fail if this is not set correctly")
 	}
+
+	DedicatedHostPoolID = os.Getenv("IBM_CONTAINER_DEDICATEDHOST_POOL_ID")
+	if ClusterName == "" {
```
Comment (lewiseevans) [inline ibm/acctest/acctest.go:885]:
```suggestion
	if DedicatedHostPoolID == "" {
```

I also wonder whether this parameter needs to be optional. Would this mean that all tests need to specify a DedicatedHostPoolID?

Comment (KerDav) [reply ibm/acctest/acctest.go]: The argument is optional by default, fixed the problems with this

File: ibm/service/kubernetes/resource_ibm_container_vpc_cluster.go [hunk 2]
```diff
@@ -450,6 +457,7 @@ func resourceIBMContainerVpcClusterCreate(d *schema.ResourceData, meta interface
 	flavor := d.Get("flavor").(string)
 	workerCount := d.Get("worker_count").(int)
 	imageSecurityEnabled := d.Get("image_security_enforcement").(bool)
+	hostPoolID := d.Get("host_pool_id").(string)
```
Comment (hkantare) [inline ibm/service/kubernetes/resource_ibm_container_vpc_cluster.go:460]: Please use d.GetOk for optioanl fields

```

if poolid, ok := d.GetOk("host_pool_id"); ok {
params.HostPoolID = poolid.(string)
}

```

Comment (hkantare) [reply ibm/service/kubernetes/resource_ibm_container_vpc_cluster.go:460]: In Read method does the API doesn't provide back the field  to d.Set

File: ibm/service/kubernetes/data_source_ibm_container_vpc_cluster.go [hunk 3]
```diff
@@ -71,6 +71,10 @@ func DataSourceIBMContainerVPCCluster() *schema.Resource {
 							Type:     schema.TypeString,
 							Computed: true,
 						},
+						"host_pool_id": {
+							Type:     schema.TypeString,
```
Comment (hkantare) [inline ibm/service/kubernetes/data_source_ibm_container_vpc_cluster.go:75]: you just defined in schema we are not setting the value

File: ibm/service/kubernetes/resource_ibm_container_vpc_cluster_test.go [hunk 4]
```diff
@@ -123,6 +123,33 @@ func TestAccIBMContainerVpcClusterImageSecuritySetting(t *testing.T) {
...
+			{
+				Config: testAccCheckIBMContainerVpcClusterDedicatedHostSetting(
+					clusterName,
+					acc.IksClusterVpcID,
+					"bx2d.4x16",
+					acc.IksClusterSubnetID,
+					acc.IksClusterResourceGroupID,
+					hostPoolID,
+				),
+				Check: resource.ComposeTestCheckFunc(
```
Comment (z0za) [inline ibm/service/kubernetes/resource_ibm_container_vpc_cluster_test.go:143]: please check if the dhost pool id is set in the workpool list as well

**Comment by hkantare** [comment]
fix the conflicts

Can we add the support for VPC worker pool resource/datasource?

File: website/docs/r/container_vpc_cluster.html.markdown [hunk 5]
```diff
@@ -159,6 +159,7 @@ Review the argument references that you can specify for your resource.
 - `image_security_enforcement` - (Optional, Bool) Set to **true** to enable image security enforcement policies in a cluster.
 - `name` - (Required, Forces new resource, String) The name of the cluster.
 - `kms_config` - (Optional, String) Use to attach a Key Protect instance to a cluster. Nested `kms_config` block has an `instance_id`, `crk_id`, `private_endpoint`.
+- `host-pool-id` - (Optional, String) If provided, the cluster will be associated with a dedicated host pool identified by this ID.
```
Comment (hkantare) [inline website/docs/r/container_vpc_cluster.html.markdown:162]: I think this shd be "host_pool_id"

**Pull request merged**